### DISCUSSION TITLE
Sphere model is fitting joints instead of links (for custom robot)

### DISCUSSION THREAD

**Discussion by jayaramreddy10** [NVlabs/curobo#56] [Software Q&A] [answered]
Hi, I was trying to build sphere model for xarm7 (which is not provided by default in curobo). Spheres are fitting joints instead of links. 

URDF : https://drive.google.com/file/d/15mpdO20_NXHT8zGIXS_X6gOnTpc5biYx/view?usp=sharing
USD :  https://drive.google.com/file/d/179Jmx-5bW3uxbX5I8bk_qkOby5i7vSJ7/view?usp=drive_link
yml file: https://drive.google.com/file/d/1rg1XP9KHyKcymljtMYX_doApf2QBbCIQ/view?usp=sharing

Screenshots of the sphere model for first 3 links. Could you please help me understand why is it modelling that way?

![link_base_xarm7](https://github.com/NVlabs/curobo/assets/111297422/ff53dcac-3950-4d50-b37f-2f7aefe0e8f7)
![link1_xarm7](https://github.com/NVlabs/curobo/assets/111297422/f6b38f79-6959-4d1e-b6d8-02d76937bc61)
![link2_xarm7](https://github.com/NVlabs/curobo/assets/111297422/b5dc5ab3-4199-49d1-a15f-06249fee21a7)

Let me know if you need further information.

**Comment by balakumar-s** [comment, marked as answer]
Not exactly sure what's happening, but you can manually move the spheres after they are generated following this video: https://curobo.org/source/tutorials/1_robot_configuration.html#robot-collision-representation

One thing to try would be to select `collisions` in `Select Mesh` drop down.

**Comment by jayaramreddy10** [comment]
Thank you, will try that.
